File: pkgs/ruby/wrappers/salesman_euclid_random_search.rb
#!/usr/bin/env ruby
# frozen_string_literal: true

lib = File.expand_path(File.join('..', 'lib'), __dir__).freeze
$LOAD_PATH.unshift lib unless $LOAD_PATH.include? lib

require 'opticontest/problems/salesman_euclid/common'
require 'opticontest/problems/salesman_euclid/random_search'

require 'yaml'

points = Opticontest::Problems::SalesmanEuclid::Common::Points.new

Integer($stdin.gets).times do
  points.add(*$stdin.gets.split.map(&method(:Float)))
end

config = Opticontest::Problems::SalesmanEuclid::RandomSearch::Config.new(
  **YAML.load_file(
    File.expand_path(
      '../../../config/salesman_euclid_random_search.yml',
      __dir__,
    ),
  ),
)

algorithm =
  Opticontest::Problems::SalesmanEuclid::RandomSearch::Algorithm.new(
    config,
    points,
  )

puts algorithm.solution.indices.join ' '

File: pkgs/python/wrappers/salesman_euclid_random_search.py
#!/usr/bin/env python3

import sys
import os

scripts_path = os.path.dirname(os.path.realpath(__file__))

sys.path.insert(0, os.path.realpath(os.path.join(scripts_path, '..')))

from opticontest.problems.salesman_euclid.common import Points
from opticontest.problems.salesman_euclid.random_search import Config, Algorithm

from yaml import load as loadYaml

count = int(input())

points = Points()

for x in range(0, count):
    points.add(*map(float, input().split()))

config = Config(
    **loadYaml(
        open(
            os.path.join(
                scripts_path,
                '../../../config/salesman_euclid_random_search.yml',
            ),
        ),
    ),
)

algorithm = Algorithm(config, points)

print(' '.join(map(str, algorithm.solution.indices)))
